src/Workspaces/Core/Portable/Workspace/Host/PersistentStorage/IPersistentStorageConfiguration.cs

```diff
@@ -58,20 +59,30 @@ private static string GetCacheDirectory()
 
         public string? TryGetStorageLocation(SolutionKey solutionKey)
         {
-            if (solutionKey.WorkspaceKind is not (WorkspaceKind.RemoteWorkspace or WorkspaceKind.RemoteTemporaryWorkspace or WorkspaceKind.Host))
-                return null;
-
             if (string.IsNullOrWhiteSpace(solutionKey.FilePath))
                 return null;
 
             // Ensure that each unique workspace kind for any given solution has a unique
             // folder to store their data in.
 
-            var cacheDirectory = GetCacheDirectory();
-            var kind = StripInvalidPathChars(solutionKey.WorkspaceKind);
-            var hash = StripInvalidPathChars(Checksum.Create(solutionKey.FilePath).ToString());
+            return Path.Combine(
+                GetCacheDirectory(),
+                SafeName(Process.GetCurrentProcess().MainModule.FileName),
+                SafeName(solutionKey.FilePath));
 
-            return Path.Combine(cacheDirectory, kind, hash);
+            static string SafeName(string fullPath)
+            {
+                var fileName = Path.GetFileName(fullPath);
+
+                // we don't want to build too long a path.  So only take 16 chars of the text we started with.
+                // However, we want to avoid collisions, so ensure we also append a safe short piece of text
+                // that is based on the full text.
```

Comment:
tweak to how we used to name things.  This way the paths we generate contain a little bit of extra information like so:

`Roslyn.sln-VGNTB48MPTe5ZsagJBTlCj0Qf0g=` instead of just `VGNTB48MPTe5ZsagJBTlCj0Qf0g=`

This is helpful for finding the right folder when you have a lot of these.  The first part is based off of the solution name, the last part is based on its entire path (in case you ahve several solutions with the same name).

Reply:
For my own knowledge why do we need to hash this? Is there a limitation on how large the string can be on the database side or do we not want to use paths because that is PII?

Reply:
yup.  there's a MAX_PATH limitation.  We can't just string together arbitrary paths of arbitrary lengths as we could easily blow past maxpath once it's all combined.  This approach effectively says that we store in this location as long as the user's appdata directory is shorter than 220 chars.   For comparison, my appdata directory is 30 chars long.  So you'd need to have a *very long* user name, or you'd need to be mounting something special to hit 220.

However, it wuold be trivial for your *solution* path or *process path* to be very long.  So we cannot just arbitrarily use those.

Reply:
`MAX_PATH` is the gift that keeps on giving